### ISSUE TITLE
"use strict"/etc directives should be labeled as such

### ISSUE THREAD

**Issue by josephfrazier** [eslint/typescript-eslint-parser#267]
**What version of TypeScript are you using?**

2.3.2

**What version of `typescript-eslint-parser` are you using?**

a294afa8158c9c088521eed72b6745eed302361c

**What code were you trying to parse?**

```js
"use strict";
"other directive";
```

**What did you expect to happen?**

I expected an AST indicating that these literals are part of a [Directive Prologue](https://tc39.github.io/ecma262/#directive-prologue).

**What happened?**

The AST doesn't differentiate between directive literals and non-directive literals. For example: https://github.com/eslint/typescript-eslint-parser/blob/f5fcc879b493138eb97b9bdfb39afc4d210b1ddf/tests/fixtures/ast/Tolerant-parse.json#L4104-L4167

For comparison, here are some other ASTs for `"octal directive"`, both of which indicate that the literal is a directive:

* Babylon: http://astexplorer.net/#/gist/29393e8fc8313aa11944f9146ceca787/0c135e856c8212bfe68702ae122261ecd4d08330
* Flow: http://astexplorer.net/#/gist/29393e8fc8313aa11944f9146ceca787/a59985860d2ee2faef66046f1d22180364a2e53b

---

Discovered in https://github.com/prettier/prettier/pull/1560#issuecomment-300300515

**Comment by soda0289** [comment]
This is not yet part of the ESTree spec. There has been a lot of discussion about it here:
https://github.com/estree/estree/issues/6
and a PR:
https://github.com/estree/estree/pull/152

We could implement the proposal and change it if its modified. But it might be better to wait for it to become official.

**Comment by josephfrazier** [comment]
Thanks for the quick response and references! That makes sense, why it's not implemented yet. Cheers!

**Comment by JamesHenry** [comment]
We will implement this if it gets accepted into ESTree

**Comment by azz** [comment]
Good news! https://github.com/estree/estree/pull/152
